src/vtcomposite.cpp

```diff
@@ -583,61 +583,89 @@ struct InternationalizeWorker : Napi::AsyncWorker
+    // required given a worldview.
+    // - If the feature has _mbx_worldview: all, return {"all"}
+    // - If there is no requested worldview "null", determine which legacy worldviews should be created
+    std::vector<std::string> worldviews_for_feature(std::string const& pval) const
     {
-        static const std::vector<std::string> legacy_worldviews{"CN", "IN", "JP", "US"};
-
         if (pval == "all")
         {
             return {"all"};
         }
 
+        // convert pval into vector of strings US,CN => {"US", "CN"}
+        // and sort ascending
+        std::vector<std::string> worldview_values;
+        std::stringstream s_stream(pval); // create string stream from the string
+        while (s_stream.good())
+        {
+            std::string substr;
+            std::getline(s_stream, substr, ','); // get first string delimited by comma
+            worldview_values.push_back(substr);
+        }
+        std::sort(worldview_values.begin(), worldview_values.end());
+
+        // result will be a vector of matching worldview values
+        std::vector<std::string> result;
+
+        // worldview: null
         if (baton_data_->worldview.empty())
         {
-            std::vector<std::string> legacy;
-            for (auto const& wv : legacy_worldviews)
-            {
-                if (pval.find(wv) != std::string::npos)
-                {
-                    legacy.push_back(wv);
-                }
-            }
-            return legacy;
+            static const std::vector<std::string> legacy_worldviews{"CN", "IN", "JP", "US"}; // MUST BE IN ALPHABETICAL ORDER
+            // {"US", "CN"} AND {"US", "IN"} => {"US"}
+            std::set_intersection(
+                worldview_values.begin(),
+                worldview_values.end(),
+                legacy_worldviews.begin(),
+                legacy_worldviews.end(),
+                std::back_inserter(result));
         }
+        // worldview: XX
         else
         {
-            if (pval.find(baton_data_->worldview) != std::string::npos)
+            if (std::find(std::begin(worldview_values), std::end(worldview_values), baton_data_->worldview) != std::end(worldview_values))
             {
-                return {pval};
+                result.push_back(baton_data_->worldview);
             }
         }
 
-        return {};
+        return result;
     }
 
-    void buildFeature(
-        vtzero::feature& const feature, 
-        property_list& const properties, 
-        vtzero::layer_builder& lbuilder
-        std::string& const worldview)
+    // create a feature from a list of properties
+    // optionally define a worldview property if provided
+    static void build_internationalized_feature(
+        vtzero::feature const& feature,
+        std::vector<std::pair<std::string, vtzero::property_value>> const& properties,
+        vtzero::layer_builder& lbuilder,
+        std::string const& worldview)
```

Comment:
nit: I had worldview at the end because I was going to add a default value (""). If not doing that, you could move it back one (between properties and lbuilder) to keep all the inputs (first 3) together and the output (lbuilder) last.

Reply:
Updated in 357db09